test_scripts/Defects/5_0/2445_SDL_doesn_t_resume_status_of_subscribtion_on_wayPoint_related_data.lua

```diff
@@ -0,0 +1,133 @@
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+local test = require("user_modules/dummy_connecttest")
+local utils = require("user_modules/utils")
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+local notifParams = {
+    wayPoints =
+    {
+        {
+            coordinate = {
+                latitudeDegrees = -90,
+                longitudeDegrees = -180
+            },
+            locationName = "Ho Chi Minh",
+            addressLines = {"182 Le Dai Hanh"},
+            locationDescription = "Toa nha Flemington",
+            phoneNumber = "1231414",
+            locationImage = {
+                value = common.getPathToFileInStorage("icon.png"),
+                imageType = "DYNAMIC"
+            },
+            searchAddress = {
+                countryName = "aaa",
+                countryCode = "084",
+                postalCode = "test",
+                administrativeArea = "aa",
+                subAdministrativeArea = "a",
+                locality = "a",
+                subLocality = "a",
+                thoroughfare = "a",
+                subThoroughfare = "a"      
+            }
+        }  
+    }
+}
+
+-- [[ Local Functions ]]
+local function cleanSessions()
+    for i = 1, common.getAppsCount() do
+      test.mobileSession[i] = nil
+    end
+    utils.wait()
+end
+
+local function pTUpdateFunc(tbl)
+    local OWgroup = {
+        rpcs = {
+            GetWayPoints = {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            },
+            SubscribeWayPoints = {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            },
+            UnsubscribeWayPoints = {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            },
+            OnWayPointChange =  {
+                hmi_levels = { "BACKGROUND", "FULL", "LIMITED" }
+            }
+        }
+    }
+    tbl.policy_table.functional_groupings["NewTestCaseGroup"] = OWgroup
+    tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.fullAppID].groups = { "Base-4", "NewTestCaseGroup" }
+end
+
+local function subscribeWayPoints()
+    local cid = common.getMobileSession():SendRPC("SubscribeWayPoints", {})
+    common.getHMIConnection():ExpectRequest("Navigation.SubscribeWayPoints")
+    :Do(function(_, data)
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",{})
+    end)
+    common.getMobileSession():ExpectResponse(cid, { success = true , resultCode = "SUCCESS" })
+    common.getMobileSession():ExpectNotification("OnHashChange")
+end
+
+local function onWayPointChange()
+    common.getHMIConnection():SendNotification("Navigation.OnWayPointChange", notifParams)
+    common.getMobileSession():ExpectNotification("OnWayPointChange", notifParams)
+end
+
+local function closeMobileSession()
```

Comment:
@TMyza According test description mobile session should be closed with unexpected disconnect of mobile device. Current function unregisters mobile application correctly without disconnect.